spec.html

```diff
@@ -7236,7 +7236,7 @@ <h1>ECMAScript Function Objects</h1>
             (lexical, strict, global)
           </td>
           <td>
-            Defines how `this` references are interpreted within the formal parameters and code body of the function. ~lexical~ means that `this` refers to the *this* value of a lexically enclosing function. ~strict~ means that the *this* value is used exactly as provided by an invocation of the function. ~global~ means that a *this* value of *undefined* is interpreted as a reference to the global object.
+            Defines how `this` references are interpreted within the formal parameters and code body of the function. ~lexical~ means that `this` refers to the *this* value of a lexically enclosing function. ~strict~ means that the *this* value is used exactly as provided by an invocation of the function. ~global~ means that a *this* value of *undefined* or *null* is interpreted as a reference to the global object, and *this* value of primitive type is wrapped by its type's wrapper object.
```

Comment:
"the global object" is incorrect anyways - it's "the [[GlobalThisValue]] of the [[GlobalEnv]] value of the current realm", which probably wouldn't help make this much clearer :-)

If `~global~` also implies sloppy mode, then I'd maybe state this like:
```suggestion
            Defines how `this` references are interpreted within the formal parameters and code body of the function. ~lexical~ means that `this` refers to the *this* value of a lexically enclosing function. ~strict~ means that the *this* value is used exactly as provided by an invocation of the function. ~global~ means that a *this* value of *undefined* or *null* is interpreted as a reference to the global object, and the *this* value is first passed to ToObject.
```

Reply:
Humm...
I agree that the expression `the global object` is incorrect. but I think `the global object of current realm` or `the [[GlobalObject]] of the current realm` would be shorter and more readable then `the [[GlobalThisValue]] of the [[GlobalEnv]] value of the current realm`. do you have any reason to use that sentence?

PS. `currentRealm.[[GlobalEnv]].[[GlobalThisValue]]` is always eqauls `currentRealm.[[GlobalObject]]`. (8.2.3 SetRealmGlobalObject is only way to change "**the global object of current realm**" as much as I know.)

anyway, 
>  the *this* value is first passed to ToObject.
this looks better then my suggestion, I'm totally support this sentence.

Reply:
> currentRealm.[[GlobalEnv]].[[GlobalThisValue]] is always eqauls currentRealm.[[GlobalObject]]

No, not always. It's either that *or* an implementation-defined object. See [InitializeHostDefinedRealm](https://tc39.github.io/ecma262/#sec-initializehostdefinedrealm), step 8.

Reply:
Did you checked next step of it? it calls SetRealmGlobalObject(and this abstract operation makes sure what I just said before.)

anyway, If you meant that `currentRealm.[[GlobalObject]]` could not be an "global object"(which spec explains at Clause 18), then I'd love to say that I missed that part.

and.. If I missed something, I'd love to say sorry to you :(

Reply:
@ENvironmentSet all browsers supply a different globalThis value from the globalObject value (a window proxy).

Reply:
@ljharb Oh, I see. thank you for comment :)
@jmdyck thank you too :))